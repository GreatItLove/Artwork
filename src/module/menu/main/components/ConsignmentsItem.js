import React from 'react';
import { ListItem } from 'material-ui/List';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import { CONSIGNMENTS_PATH } from '../../../../config/routesPatch';

const propTypes = {
  style: PropTypes.object.isRequired,
  styleChild: PropTypes.object.isRequired
};

/**
 * @function ConsignmentsItem
 * @param {Object} style
 * @param {Object} styleChild
 * @returns {XML}
 */
function ConsignmentsItem({ style, styleChild }) {
  return (
    <ListItem
      key={CONSIGNMENTS_PATH}
      primaryText="Consignments"
      innerDivStyle={style}
      leftIcon={<i className="material-icons">local_shipping</i>}
      primaryTogglesNestedList={true}
      nestedItems={[
        <ListItem
          key="Overview"
          primaryText="Overview (View All)"
          innerDivStyle={styleChild}
          containerElement={<NavLink to={CONSIGNMENTS_PATH} exact />}
        />,
        <ListItem key="ConsignedOut" innerDivStyle={styleChild} primaryText="Consigned Out" />
      ]}
    />
  );
}

ConsignmentsItem.propTypes = propTypes;

export default ConsignmentsItem;
