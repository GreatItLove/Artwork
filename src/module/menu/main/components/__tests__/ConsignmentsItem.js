import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ConsignmentsItem from '../ConsignmentsItem';

describe('ConsignmentsItem', () => {
  it('Snapshot of ConsignmentsItem', () => {
    const wrapper = shallow(<ConsignmentsItem style={{}} styleChild={{}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
