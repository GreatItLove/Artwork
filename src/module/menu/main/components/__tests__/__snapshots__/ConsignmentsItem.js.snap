// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsignmentsItem Snapshot of ConsignmentsItem 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  innerDivStyle={Object {}}
  insetChildren={false}
  leftIcon={
    <i
      className="material-icons"
    >
      local_shipping
    </i>
  }
  nestedItems={
    Array [
      <ListItem
        autoGenerateNestedIndicator={true}
        containerElement={
            <NavLink
                activeClassName="active"
                ariaCurrent="true"
                exact={true}
                to="/consignments"
            />
        }
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        innerDivStyle={Object {}}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Overview (View All)"
        primaryTogglesNestedList={false}
        secondaryTextLines={1}
    />,
      <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        innerDivStyle={Object {}}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Consigned Out"
        primaryTogglesNestedList={false}
        secondaryTextLines={1}
    />,
    ]
  }
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Consignments"
  primaryTogglesNestedList={true}
  secondaryTextLines={1}
/>
`;
